### PR TITLE
Added third metric

### PR DESCRIPTION
**Pull request by Jbaren01**
-updated the system-data react component to conditionally render a "Pending Duration" card
-updated backend routes to grab the pending duration property available from kafkajs

### CODE REVIEW

File: app/components/SystemData.jsx [hunk 1]
```diff
@@ -1,36 +1,66 @@
...
-      sum === 0 && numOfDataPoints === 0
-        ? 'None'
-        : (sum / numOfDataPoints).toFixed(2);
+             // const avg =
+             //   Math.round(
+             //     (this.props.data.sum / this.props.data.numOfDataPoints) * 100
+             //   ) / 100; // Round to 2 decimals
+             const { title } = this.props;
+             // if title === 'data size' do the following:
+             if (title === 'Data Size' || title === 'Processing Time') {
```
Comment (Shreshth3) [inline app/components/SystemData.jsx]: I think this feature can be implemented without this messy conditional logic. What if we track pendingDuration in the same way we track the other metrics? That way we can display average, smallest, and largest. This also allows us to keep our code DRY.

**Review by Shreshth3 — CHANGES_REQUESTED** [review]

File: app/components/SystemData.jsx (hunk 1 above)
Comment (Jbaren01) [reply app/components/SystemData.jsx]: ok- we tried this and were successful, however, the algorithm is not working correctly for "pending duration"

File: app/components/Dashboard.jsx [hunk 2]
```diff
@@ -37,32 +43,53 @@ class Dashboard extends Component {
...
+                                        dataSize.smallest,
+                                        data.size
+                                      ),
+                                      largest: Math.max(
+                                        dataSize.largest,
+                                        data.size
+                                      ),
+                                    },
+                                    pendingDuration: {
+                                      sum: data.pendingDuration,
```
Comment (Shreshth3) [inline app/components/Dashboard.jsx]: Here you're only setting sum to be the new value. For example, let's say I sent a 1 up the Web Socket. Then sum = 1. Now if I sent another 1, we set sum to be data.pendingDuration which would be 1. But, the correct answer is 2. You are forgetting to add the previous values (from before the current value). Maybe look at my implementation to think about how you can do that.

File: app/components/Dashboard.jsx [hunk 3]
```diff
@@ -37,32 +43,53 @@ class Dashboard extends Component {
...
+                                      numOfDataPoints:
+                                        pendingDuration.numOfDataPoints + 1,
+                                      smallest: 0,
```
Comment (Shreshth3) [inline app/components/Dashboard.jsx]: Why are these both set to 0?

File: app/components/SystemData.jsx [hunk 4]
```diff
@@ -11,12 +12,30 @@ class SystemData extends Component {
     //     (this.props.data.sum / this.props.data.numOfDataPoints) * 100
     //   ) / 100; // Round to 2 decimals
     const { title } = this.props;
-    const { sum, numOfDataPoints, smallest, largest } = this.props.data;
+    const { sum, numOfDataPoints, smallest, largest, name } = this.props.data;
```
Comment (Shreshth3) [inline app/components/SystemData.jsx]: Hmmmm why is name needed here? It doesn't look like data will have a property called name

File: app/components/Dashboard.jsx [hunk 5]
```diff
@@ -2,10 +2,10 @@ import React, { Component } from 'react';
 import SystemData from './SystemData';
```
Comment (Shreshth3) [inline app/components/Dashboard.jsx:2]: Make sure you merge your code with the new changes that have happened on the main branch! Check adding-features.md to see how to do this

File: app/components/SystemData.jsx [hunk 6]
```diff
@@ -11,12 +12,30 @@ class SystemData extends Component {
...
     const avg =
       sum === 0 && numOfDataPoints === 0
         ? 'None'
         : (sum / numOfDataPoints).toFixed(2);
 
+    //  return (
```
Comment (Shreshth3) [inline app/components/SystemData.jsx]: Let's go ahead and delete this code since it's commented out anyway. Just to keep things cleaner

**Review by Shreshth3 — APPROVED** [review]

**Comment by Shreshth3** [comment]
Great work!

**Pull request merged**